firestore-translate-text/functions/src/index.ts

```diff
@@ -79,13 +82,78 @@ export const fstranslate = functions.handler.firestore.document.onWrite(
     } catch (err) {
       logs.error(err);
     }
-  }
-);
+  });
+
+export const fstranslatebackfill = functions.tasks
+  .taskQueue()
+  .onDispatch(async (data: any) => {
+    const runtime = getExtensions().runtime();
+    if (!config.doBackfill) {
+      await runtime.setProcessingState(
+        "PROCESSING_COMPLETE",
+        "Existing documents were not translated because backfill was not requested."
+      );
+      return;
+    }
+    const offset = (data["offset"] as number) ?? 0;
+    const pastSuccessCount = (data["successCount"] as number) ?? 0;
+    const pastErrorCount = (data["errorCount"] as number) ?? 0;
+
+    const snapshot = await admin
+      .firestore()
+      .collection(process.env.COLLECTION_PATH)
+      .offset(offset)
+      .get();
+    const translations = await Promise.allSettled(
+      snapshot.docs.map(handleExistingDocument)
+    );
+    const newSucessCount =
+      pastSuccessCount +
+      translations.filter((p) => p.status === "fulfilled").length;
+    const newErrorCount =
+      pastErrorCount +
+      translations.filter((p) => p.status === "rejected").length;
+
+    if (snapshot.size == DOCS_PER_BACKFILL) {
+      const queue = getFunctions().taskQueue(
+        "fstranslatebackfill",
+        process.env.EXT_INSTANCE_ID
+      );
+      await queue.enqueue({
+        offset: offset + DOCS_PER_BACKFILL,
+        successCount: newSucessCount,
+        errorCount: newErrorCount,
+      });
+    } else {
+      logs.backfillComplete(newSucessCount, newErrorCount);
+      if (newErrorCount == 0) {
+        runtime.setProcessingState(
+          "PROCESSING_COMPLETE",
+          `Successfully translated ${newSucessCount} documents.`
+        );
+      } else if (newErrorCount > 0 && newSucessCount > 0) {
+        runtime.setProcessingState(
+          "PROCESSING_WARNING",
+          `Successfully translated ${newSucessCount} documents, ${newErrorCount} errors. See function logs for specific error messages.`
+        );
+      }
+      if (newErrorCount > 0 && newSucessCount == 0) {
```

Comment:
Maybe we don't even need a condition around this one if we _return_ out of the other 2 setProcessingState ones.
Then it would also function as a catchall if something weird happened with DOC_PER_BACKFILL and we were counting down instead of up. It would also guarantee that we not leave the processing status as 'processing' forever. I mean that shouldn't happen, but if it does, we have a reasonable fallback that reports the error.